index.rst

```diff
@@ -42,20 +42,163 @@
+* Should the OCPS check that the data really are in the OODS before accepting the command?
+  There is nothing in the system to prevent an observing script from sending the process command before ingestion.
+* Should the observing script wait for the event from the OCPS indicating that the data were processed to completion before taking more data?
+  Generally it should not wait unless it is interested in the telemetry.
+* If data fail processing for some reason should the observing script be expected to know about it?
+  Should the completion event indicate whether the pipeline completed without error?
+* If data are processed but some QA metric fails, should it be included in the co-add?
+  Failing QA will result in the output data being stored in the Butler repository and would likely result in good pipeline exit status.
+* Should metrics be published to the SQuaSH system in addition to being published as telemetry?
+  Given the transient nature of the summit data products and the presence of the metrics in the EFD via telemetry, it may not be necessary to also store them in SQuaSH.
+* Does SAL require that all metrics that could possibly be reported from a pipeline be declared in advance?
+  Or can metrics be issued as a JSON document and allow any content?
+  Does LOVE work in such a dynamic environment?
+
+Camera Diagnostic Cluster
+=========================
+
+The diagnostic cluster is currently designed to be used by the camera system for their own analysis.
+The command issued by the CCS to trigger processing could be anything and could include an explicit ingest of the data files into a diagnostic-cluster-specific butler followed by calls to DM Pipeline code.
+If the analysis being performed is substantially the same as that being performed by the OCPS then there is scope for the systems to be combined into a single system.
 
 .. note::
 
-   **This technote is not yet published.**
+   If the diagnostic cluster is used as the compute environment for OCPS then it will be necessary for the single-frame processing pipelines that are executed to always generate the JPEGs required for the summit visualization system.
+
+Conclusions
+===========
+
+A system is proposed to allow observing scripts to initiate data processing of data taken by that observing script.
+It uses a well-defined interface to make processing requests and can wait for completion events or continue to take additional data.
+
+Appendix A: Future Options
+==========================
+
+In the above design, the observing script makes individual requests to the processing system.
+An alternative approach can also be considered whereby the OCPS listens directly to the OODS and automatically processes the data without being commanded to do so by the observing script.
+The OCPS would issue an event when data are sent for processing and an event when processing has completed.
+For standard survey observing this approach is more than adequate and simplifies observing scripts.
```

Comment:
I don't understand why this simplifies observing scripts.  Either you send a message with the pipeline name to OCPS, or you set a metadata field with the pipeline name.